### DISCUSSION TITLE
Cannot upload root disc - convert datadisk to root disk

### DISCUSSION THREAD

**Discussion by AlexanderKgr** [apache/cloudstack#9910] [Q&A] [answered]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Uploading volume cannot be chosen as root disk. It is uploaded as datadisk and after that it is not possible to convert it to root disc
~~~

##### CLOUDSTACK VERSION

~~~
4.19.1.1
~~~

##### CONFIGURATION

Ubuntu 22.04 server edition

##### OS / ENVIRONMENT

Cloudstack 4.19.1.1 installed on ubuntu 22.04 server edition with xcp-ng as hypervisor

##### STEPS TO REPRODUCE

~~~
Uploading volume cannot be chosen as root disk. It is uploaded as datadisk and after that it is not possible to convert it to root disc
~~~

![{BB8825F7-4628-41F6-86EB-9D84CDBA6591}](https://github.com/user-attachments/assets/7697bf37-3122-430b-8376-759bb0f93ae6)

![{815F61C7-20B8-4B16-A3CC-2B3FD8A6D72C}](https://github.com/user-attachments/assets/0949b12f-73d8-4427-bbb2-61939f7d270b)

##### EXPECTED RESULTS

~~~
There could be an  option to select type of disc and it could have been uploaded as a root disc type
~~~

##### ACTUAL RESULTS

~~~
There is no option to select the type of the disc. Disc is uploaded as datadisk type and it is not possible to attach it to a vm.
~~~

**Comment by rajujith** [comment]
@AlexanderKgr You can upload this file as a template? When you deploy an instance from template it gets copied to the desired ROOT volume.

**Comment by AlexanderKgr** [comment]
> @AlexanderKgr You can upload this file as a template? When you deploy an instance from template it gets copied to the desired ROOT volume.

i don't want to create new instance. i just want to replace the volume. In previous versions of cloudstack it was possible.

**Comment by weizhouapache** [comment, marked as answer]
@AlexanderKgr 
to be frank, I do not think it was possible to upload a volume as root disk, because the Uploaded volume is stored on secondary storage, it is not attached to any vm.

you can try
- attach the new volume to the vm as datadisk
- if the vm is not running, start the vm and then stop it. The Uploaded volume will be copied from secondary storage to same storage pool as the root volume.
- detach the new volume
- detach the ROOT volume of the vm you want to replace
- attach the new volume to the vm as ROOT disk (deviceId=0)
